### DISCUSSION TITLE
Debug messages in mode_analysis report (nul) lines but extracting fine

### DISCUSSION THREAD

**Discussion by mpm-meto** [dtcenter/METplus#1335] [Statistical Computation] [answered]
I am running mode_analysis and getting the following debug message. It does not appear to affect the output but I know it used to work in previous versions. It certainly is disconcerting! I've attached a file for you to try and the command run.  

The message I get is this. This is with MET10.0.0.

DEBUG 2: Total mode lines read = (nul)
DEBUG 2: Total mode lines kept = (nul)

Running on RHEL7 virtual workstation.
```
	mode_analysis \
	    -lookin ${datapath}/input.txt \
	    -summary \
	    -centroid_lat_min ${minlat} \
	    -centroid_lat_max ${maxlat} \
	    -column AREA -column LENGTH -column WIDTH -column AREA -column CURVATURE -column INTENSITY_90 \
	    -fcst -single \
	    -dump_row ${outpath}/job_summary_wind_speed_${valid}Z_${fcr}_${outid}_fcst.txt \
	    -out      ${outpath}/job_summary_wind_speed_${valid}Z_${fcr}_${outid}_fcst.out -v 3
```

[input.txt](https://github.com/dtcenter/METplus/files/7743910/input.txt)

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi @mpm-meto. I took at look at the GitHub issue you created in the METplus repository, did some digging, and decided it'd be better to handle this via Discussions. So I converted your issue into this discussion.

The short story is that this issue has already been fixed on the develop branch by pull request dtcenter/MET#1915 which fixed issue dtcenter/MET#1912. Here's the actual modified code:
   https://github.com/dtcenter/MET/commit/ab4c7314f13494fd693f31b35bd9d314246ed7d9#

I don't think this requires an actual bugfix because the issue is confined to the formatting of a log message, rather than any actual problem in the output.

Here's the longer story...

First, thanks for sending along some sample data and command to demonstrate the issue. As I understand it, the issue is that MODE-Analysis is not reporting an accurate count of the number of lines being read... and the number being retained (i.e. surviving the filtering step):
```
DEBUG 2: Total mode lines read = (nul)
DEBUG 2: Total mode lines kept = (nul)
```
I used the data you sent and was able to replicate this behavior using MET version 10.0.0. I was also able to replicate this behavior using MET version 9.1.3. So it's an issue that's been around for a while. However there are 2 pieces of good news:
1. The problem has already been fixed in the develop branch in preparation for version 10.1.0. Testing that version I see:
```
DEBUG 2: Total mode lines read = 8048
DEBUG 2: Total mode lines kept = 2626
```
2. The problem in version 10.0.0 and 9.1.3 are confined to printing the log messages. By that I mean that the number of lines in the "-dump_row" file that's created actually do match the "Total mode lines kept" reported in the develop branch.
